src/GafferSceneUI/ContextAlgo.cpp

```diff
@@ -50,6 +51,43 @@ namespace
 InternedString g_expandedPathsName( "ui:scene:expandedPaths" );
 InternedString g_selectedPathsName( "ui:scene:selectedPaths" );
 
+bool expandWalk( const ScenePlug::ScenePath &path, const ScenePlug *scene, size_t depth, PathMatcher &expanded, PathMatcher &leafPaths )
+{
+	bool result = false;
+
+	ConstInternedStringVectorDataPtr childNamesData = scene->childNames( path );
+	const std::vector<InternedString> &childNames = childNamesData->readable();
+
+	if( childNames.size() )
+	{
+		result |= expanded.addPath( path );
```

Comment:
Lifting this out of the conditional makes `SceneViewTest.testExpandPaths` fail.

```
FAIL: testExpandSelection (GafferSceneUITest.SceneViewTest.SceneViewTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/andrewk/apps/gaffer/0.33.2.0dev/cent7.x86_64/cortex/9/gaffer/python/GafferSceneUITest/SceneViewTest.py", line 126, in testExpandSelection
    self.assertEqual( getExpandedPaths(), set( [ "/", "/A", "/A/C" ] ) )
AssertionError: Items in the first set but not the second:
'/A/C/E'
'/A/C/D'
'/A/B'
----------------------------------------------------------------------
```

If I'm reading the comment correctly, its explicitly saying that objects that don't have children can never be considered expanded.

```
# do a recursive expansion from the root. all leafs should be selected.
# leaf items should not be expanded, because there are no children to show.

setSelection( [ "/A" ] )
setExpandedPaths( [ "/" ] )

view.expandSelection( depth = 3 )
self.assertEqual( getExpandedPaths(), set( [ "/", "/A", "/A/C" ] ) ) # this line fails
self.assertEqual( getSelection(), set( [ "/A/B", "/A/C/D", "/A/C/E" ] ) )
```